### PR TITLE
Show point doubling correctly

### PR DESCRIPTION
**Pull request by abalkin**
See issue #1.

### CODE REVIEW

File: src/components/sharygin-cubic-graph.js [hunk 1]
```diff
@@ -18,8 +18,16 @@ const getScaledPointsList = pointsDict => {
 };
 
 const getLongLine = (x1, y1, x2, y2) => {
-  const dx = x1 - x2;
-  const dy = y1 - y2;
+  let dx, dy;
+  if (x1 !== x2) {
```
Comment (ocfnash) [inline src/components/sharygin-cubic-graph.js]:
```suggestion
  if (x1 !== x2 && y1 !== y2) {
```
The cubic actually intersects some vertical lines so I think we need to check both coordinates here.

File: src/components/sharygin-cubic-graph.js [hunk 2]
```diff
@@ -18,8 +18,16 @@ const getScaledPointsList = pointsDict => {
...
+    dx = x1 - x2;
+    dy = y1 - y2;
+  }
+  else {
+    const c = 1 - (x1 + y1) * (x1 + y1 + 1)
+    dx = c - y1 * (2 * y1 + 1)
+    dy = c + x1 * (2 * x1 + 1)
```
Comment (ocfnash) [inline src/components/sharygin-cubic-graph.js]:
```suggestion
    dy = -c + x1 * (2 * x1 + 1)
```
Tiny sign error here I think.

**Comment by ocfnash** [comment]
Thanks for this!

I have just updated the version embedded [here](http://olivernash.org/2016/12/18/sharygins-group-of-triangles/index.html) to use your fix.

Really appreciate the help :-)

**Comment by abalkin** [comment]
Ideally we should compare the indices of the sample points rather than their coordinates but that would require some refactoring.

Also, I find it very difficult to hit the case P=Q in practice. I'll open a separate issue to ass a toggle button to force P=Q case.

**Comment by ocfnash** [comment]
> Ideally we should compare the indices of the sample points rather than their coordinates but that would require some refactoring.

Yes I had exactly the same thoughts. Unfortunately I don't have the appetite for the refactoring at the moment!


> Also, I find it very difficult to hit the case P=Q in practice. I'll open a separate issue to ass a toggle button to force P=Q case.

It is indeed hard to trigger the `P = Q` case but I'm not convinced we need a feature to force `P = Q`. After all, it is almost impossible for the user to notice the difference between a near match and a perfect match. It definitely _was_ worth fixing the bug but I feel things are now in a reasonable state.

**Pull request merged**